client/lib/viewport/README.md

```diff
@@ -42,28 +42,67 @@ class MyComponent extends React.Component {
 	};
 
 	componentDidMount() {
-		addIsDesktopListener( this.sizeChanged );
+		this.unsubscribe = addIsDesktopListener( this.sizeChanged );
 	}
 
 	componentWillUnmount() {
-		removeIsDesktopListener( this.sizeChanged );
+		if ( this.unsubscribe ) {
+			this.unsubscribe();
+		}
 	}
 }
 ```
 
+It also comes with React helpers that help in registering a component to breakpoint changes.
+
+Using a hook:
+
+```js
+import { useMobileBreakpoint } from 'lib/viewport/react-helpers';
+
+export default function MyComponent( props ) {
+	const isMobile = useMobileBreakpoint();
+	return <div>Screen size: { isMobile ? 'mobile' : 'not mobile' }</div>;
+}
+```
+
+Using a higher-order component:
+
+```js
+import { withMobileBreakpoint } from 'lib/viewport/react-helpers';
+
+class MyComponent extends React.Component {
+	render() {
+		const { isBreakpointActive: isMobile } = this.props;
+		return <div>Screen size: { isMobile ? 'mobile' : 'not mobile' }</div>;
+	}
+}
+
+export default withMobileBreakpoint( MyComponent );
+```
+
 ### Supported methods
 
-- `isWithinBreakpoint( breakpoint )`: Whether the current screen size matches the breakpoint
-- `isMobile()`: Whether the current screen size matches a mobile breakpoint (<480px)
-- `isDesktop()`: Whether the current screen size matches a desktop breakpoint (>960px)
-- `addWithinBreakpointListener( breakpoint, listener )`: Register a listener for size changes that affect the breakpoint
-- `removeWithinBreakpointListener( breakpoint, listener )`: Unregister a previously registered listener
-- `addIsMobileListener( breakpoint, listener )`: Register a listener for size changes that affect the mobile breakpoint (<480px)
-- `removeIsMobileListener( breakpoint, listener )`: Unregister a previously registered listener
-- `addIsDesktopListener( breakpoint, listener )`: Register a listener for size changes that affect the desktop breakpoint (>960px)
-- `removeIsDesktopListener( breakpoint, listener )`: Unregister a previously registered listener
+- `isWithinBreakpoint( breakpoint )`: Whether the current screen size matches the breakpoint.
+- `isMobile()`: Whether the current screen size matches a mobile breakpoint (<480px).
+- `isDesktop()`: Whether the current screen size matches a desktop breakpoint (>960px).
+- `addWithinBreakpointListener( breakpoint, listener )`: Register a listener for size changes that affect the breakpoint. Returns the unsubscribe function.
+- `addIsMobileListener( listener )`: Register a listener for size changes that affect the mobile breakpoint (<480px). Returns the unsubscribe function.
+- `addIsDesktopListener( listener )`: Register a listener for size changes that affect the desktop breakpoint (>960px). Returns the unsubscribe function.
```

Comment:
One more thing: `subscribeWithinBreakpoint` might be a better name than one that contains `addListener`. `addListener` suggests the other listener pattern, used commonly in DOM, the one that we don't use here :wink: What do you think?

Reply:
Sounds good! I'll keep the `is` to make it clear we're talking about subscribing to a Boolean: `subscribeIsWithinBreakpoint`, `subscribeIsMobile`, `subscribeIsDesktop`.